bridge/bridge.go

```diff
@@ -244,6 +273,78 @@ func (b *Bridge) add(containerId string, quiet bool) {
 		b.services[container.ID] = append(b.services[container.ID], service)
 		log.Println("added:", container.ID[:12], service.ID)
 	}
+
+	// if swarm container belongs to swarm mode service, publish VIP services
+	if swarmServiceName, ok := container.Config.Labels["com.docker.swarm.service.name"]; ok {
+		filters := map[string][]string{"name": {swarmServiceName}}
+		services, err := b.docker.ListServices(dockerapi.ListServicesOptions{Filters: filters})
+		if err != nil {
+			log.Println("error listing swarm services, wont register VIP service", err)
+		} else if len(services) == 1 { // container cannot belong to no or more than one service
+			if services[0].Spec.EndpointSpec != nil {
+				mode := services[0].Spec.EndpointSpec.Mode
+				if mode == swarm.ResolutionModeVIP { // endpoint should be VIP
+					if (len(services[0].Endpoint.VirtualIPs) > 0) {
+						b.registerSwarmVipServices(services[0])
+					}
+				}
+			}
+
+		}
+	}
+}
+
+func (b *Bridge) SyncSwarmServices() {
+	// get existing swarm services
+	servicefilters := map[string][]string{}
+	swarmServices, err := b.docker.ListServices(dockerapi.ListServicesOptions{Filters: servicefilters})
+	if err != nil {
+		log.Println("error listing swarm services, wont register VIP service", err)
+	}
+
+	// get register services
+	myservices, err := b.registry.Services()
+	if err != nil {
+		log.Println("error listing registry services", err)
+	}
+
+	// remove register services doesn't exist in swarm services
+	for _, myservice := range myservices {
+		for _, tag := range myservice.Tags {
+			if tag == "vip-outside" {
```

Comment:
According to documentation only consul backend support tags currently, so this will not work with other backends.